// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

package feedwire;

import "components/feed/core/proto/v2/wire/capabilities_debug_data.proto";
import "components/feed/core/proto/v2/wire/chrome_feed_response_metadata.proto";
import "components/feed/core/proto/v2/wire/content_lifetime.proto";
import "components/feed/core/proto/v2/wire/data_operation.proto";
import "components/feed/core/proto/v2/wire/eventid.proto";
import "components/feed/core/proto/v2/wire/info_card.proto";
import "components/feed/core/proto/v2/wire/server_experiment_data.proto";

option optimize_for = LITE_RUNTIME;

message FeedResponse {
  repeated DataOperation data_operation = 1;
  optional FeedResponseMetadata feed_response_metadata = 2;
}
message FeedResponseMetadata {
  optional int64 response_time_ms = 1;
  optional EventIdMessage event_id = 2;
  optional bool pinned_content_fulfilled = 4;
  optional ServerExperimentData server_experiment_data = 5;
  optional ContentLifetime content_lifetime = 6;
  optional InfoCardServingInfo info_card_serving_info = 7;
  optional CapabilitiesDebugData capabilities_debug_data = 8;
  optional ChromeFeedResponseMetadata chrome_feed_response_metadata = 326233599;
}
